pyscriptjs/src/exceptions.ts

```diff
@@ -2,20 +2,49 @@ const CLOSEBUTTON = `<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'
 
 type MessageType = "text" | "html";
 
+/*
+These error codes are used to identify the type of error that occurred.
+The convention is:
+* PY0 - errors that occur when fetching
+* PY1 - errors that occur in config
+* PY9 - Deprecation errors
+*/
+
+export enum ErrorCode {
+  GENERIC = "PY0000", // Use this only for development then change to a more specific error code
+  FETCH_ERROR = "PY0001",
+  FETCH_NAME_ERROR = "PY2002",
+  // Currently these are created depending on error code received from fetching
+  FETCH_UNAUTHORIZED_ERROR = "PY0401",
+  FETCH_FORBIDDEN_ERROR = "PY0403",
+  FETCH_NOT_FOUND_ERROR = "PY0404",
+  FETCH_SERVER_ERROR = "PY0500",
+  FETCH_UNAVAILABLE_ERROR = "PY0503",
+  BAD_CONFIG = "PY1000",
+  TOP_LEVEL_AWAIT = "PY9000"
+}
```

Comment:
the FETCH_* errors doesn't seem to match what's described in the doc, they still use the old `PY2...` bucket